shell/packages/sandstorm-ui-grainlist/grainlist-client.js

```diff
@@ -123,12 +128,14 @@ SandstormGrainListPage.bulkActionButtons = function (showTrash) {
         },
 
         onClicked: function (ownedGrainIds, sharedGrainIds) {
-          ownedGrainIds.forEach((grainId) => {
+          const grainIds = ownedGrainIds.concat(sharedGrainIds);
+
+          grainIds.forEach((grainId) => {
```

Comment:
I agree with this.

Maybe the `deleteGrain` method should be extended to automatically delete any owner-owned API tokens? Then you call `deleteGrain` if the grain is owned by the user and `forgetGrain` otherwise, but don't have to worry about the "both" case.

Or maybe these two methods should be merged into one with the server figuring out which one applies.